src/components/control/date.test.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import {
+  render,
+  screen,
+} from '@testing-library/react';
+import user from '@testing-library/user-event';
+import Date from './date.jsx';
+
+describe('<Date />', () => {
+  describe('onActivate API', () => {
+    it('toggles date to active when user clicks on date control', () => {
+      const onActivate = jest.fn();
+      render(<>
+        <Date id="test-id" onActivate={onActivate} />
+      </>);
+
+      user.click(screen.getByRole('textbox'));
+      expect(onActivate).toHaveBeenCalledWith(undefined);
+    });
+
+    it('toggles date and sends value at time when user clicks on date control', () => {
+      const onActivate = jest.fn();
+      render(<>
+        <Date id="test-id" value="2022-01-01" onActivate={onActivate} />
+      </>);
+
+      user.click(screen.getByRole('textbox'));
+      expect(onActivate).toHaveBeenCalledWith(new window.Date('2022-01-01T00:00:00'));
+    });
+
+    it('toggles the date to active when user tabs into Date Control', () => {
+      const onActivate = jest.fn();
+      render(<>
+        <button aria-label="pre input button" />
+        <label id="test-id-label">Date</label>
+        <Date id="test-id" onActivate={onActivate} />
+      </>);
+
+      // User has Focused on Element Before Date Input
```

Comment:
IIRC, I _think_ the "first" element focused on mount is the document body. You could just tab into the date.

Reply:
Yeah. I wanted to walk through the accessibility story here. Especially with Keyboard focused users.